src/main/webapp/app/exercises/programming/shared/code-editor/file-browser/code-editor-file-browser-file.component.html

```diff
@@ -19,12 +19,14 @@
             (keyup.esc)="clearRenamingNode($event)"
             class="list-group-item__input"
     /></span>
+    <!-- File badges -->
+    <jhi-file-browser-badge *ngFor="let badge of badges" [badge]="badge" [onColoredBackground]="item.checked || hasError || hasChanges"></jhi-file-browser-badge>
     <!-- File level actions -->
-    <span class="file-icons" [ngClass]="item.checked ? 'text-white' : 'primary'">
-        <button [disabled]="disableActions" (click)="setRenamingNode($event)" class="btn btn-small">
+    <span class="file-icons" [ngClass]="item.checked ? 'text-white' : 'primary'" *ngIf="!disableActions">
```

Comment:
Note for code reviewers: We now hide the file actions completely if they are not available: A tutor doesn't need to see a rename button they can't press anyway.